src/polycubed/src/utils/veth_peer.cpp

```diff
@@ -0,0 +1,341 @@
+#include <stdlib.h>
+#include "veth_peer.h"
+
+#define NETNS_RUN_DIR "/var/run/netns"
+
+/* useful for memory paging */
+#define SIZE_ALIGN 8192
+
+VethPeer::VethPeer() {
+  resetPeer();
+}
+
+VethPeer::~VethPeer() {}
+
+void VethPeer::resetPeer() {
+  name_ = "";
+  status_ = false;
+  ns_ = "root";
+  mac_ = "";
+  ip_ = "";
+  prefix_ = -1;
+}
+
+/* set functions */
+void VethPeer::setName(const std::string &name) {
+  name_ = name;
+}
+
+void VethPeer::setStatus(const bool status) {
+  /* status false = DOWN; status true = UP; */
+  if (status_ == status) {
+    return;
+  }
+  if (ns_ != "root") {
```

Comment:
Using "root" to check if this is the root namespace is wrong, it is possible for a namespace to be called "root".

The can say that if ns == "" then the interface is in the root namespace.